src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
If you really want to benchmark this you want to set the function up in isolation and benchmark it with criterion. Of course you won't see it in the noise of _16 seconds_ of running the compiler. But performance is not some magical thing that we can only assess through measurement. It's obvious that the new version is doing more work than the old one.